1-bootstrap/README.md

```diff
@@ -1,32 +1,381 @@
 
-You will need a Google Cloud project with [Secret Manager](https://cloud.google.com/security/products/secret-manager) to store your git credentials, throughout the documentation this will be referenced as `$GIT_SECRET_PROJECT`.
+You can optionally specify an existing Cloud Storage bucket to store logs from:
+
+- Build logs
+- Terraform state bucket
+
+The bucket will use the KMS Key provided to encrypt the content. In this case, the code will grant the Storage Service Agent:
+
+   - Cloud KMS CryptoKey Encrypter: `roles/cloudkms.cryptoKeyEncrypter`
+   - Cloud KMS CryptoKey Decrypter: `roles/cloudkms.cryptoKeyDecrypter`
+
+If a Key is not provided, a new one will be created at the same project to encrypt the content.
+
+### VPC Service Controls (VPC-SC)
+
+This module supports deployment within a VPC-SC perimeter.
+
+__Important:__ The Cloud Build project (seed project) __cannot__ be within the perimeter. This is because the module creates new projects, and errors will occur when accessing services (e.g., enabling APIs) before the new project is added to the perimeter.
+
+This module does not create the Service Perimeter or Access Level. However, it can add projects to the Service Perimeter, create directional rules, and add identities to the Access Level.
+
+To enable VPC-SC integration, you must provide the following:
+
+- An existing Access Level name.
+- An existing Service Perimeter name.
+- The deployment mode (`DRY_RUN` or `ENFORCED`).
+
+The identity deploying the module must be a member of the specified Access Level.
+
+- __IAM Roles:__ The identity deploying the module requires the following IAM role at the organization level:
+
+   - __Access Context Manager Admin:__ `roles/accesscontextmanager.policyAdmin`
+
+   ```bash
+   gcloud organizations add-iam-policy-binding YOUR_ORGANIZATION_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
+   --role="roles/accesscontextmanager.policyAdmin"
+   ```
+
+### Private Worker Pool Requirements
+
+A private worker pool is required to run within a VPC-SC perimeter.
+
+A sample Worker Pool configuration without external IP, peered network, and NAT is provided in the `test/setup/modules/private_workerpool/` folder. This sample includes a project, a Private Worker Pool in its own project, a peered VPC with a NAT VM that allows internet egress for external dependencies, and the necessary firewall rules and configurations.
+
+The same pool can be used across multiple steps. Reserving a wider IP range allows for more concurrent builds. A /24 range supports 254 hosts.
+
+- __IAM Roles:__ The identity deploying the module must have the following IAM roles on the Private Worker Pool project:
+
+   - __Cloud Build WorkerPool User:__ `roles/cloudbuild.workerPoolUser`
+   - __Project IAM Admin:__ `roles/resourcemanager.projectIamAdmin`
+
+   ```bash
+   gcloud projects add-iam-policy-binding YOUR_WORKER_POOL_PROJECT_ID \
+   --member="serviceAccount:YOUR_SERVICE_ACCOUNT_EMAIL" \
```

Comment:
maybe use a ` export SERVICE_ACCOUNT_EMAIL='YOUR_SERVICE_ACCOUNT_EMAIL'`  at the start and allow for just copy past the commands